ipie/addons/thermal/estimators/generic.py

```diff
@@ -0,0 +1,120 @@
+import plum
+import numpy
+from ipie.hamiltonians.generic import GenericRealChol, GenericComplexChol
+
+
+@plum.dispatch
```

Comment:
Aren't these identical to the non thermal functions? P should have a type hint.

Reply:
I think the non-thermal `local_energy_generic_cholesky` (https://github.com/JoonhoLee-Group/ipie/blob/693f3dbf861f4ee75f9f923a5f7c26959864e85f/ipie/estimators/generic.py#L55) is wrong for the `GenericComplexChol` case. The value returned disagrees with the energy computed from ERIs. But I also don't think `local_energy_generic_cholesky` is used in the non-thermal code.

Reply:
There's a bug for complex ERIs, or rather the linked function assumes real integrals. Could you open an issue?